kmod/src/xattr.c

```diff
@@ -481,6 +486,21 @@ void scoutfs_xattr_init_totl_key(struct scoutfs_key *key, u64 *name)
 	key->skxt_c = cpu_to_le64(name[2]);
 }
 
+static int parse_worm_name(const char *name, int name_len)
+{
+	char worm_name[] = "scoutfs.worm.v1_expiration";
```

Comment:
It should only be checking the part of the name string after the tags.   It shouldn't bake in the scoutfs prefix and only assume that there's only the worm tag.   I'd look for the last dot  (strchr()?) and then compare from after the dot to the end with v1_expiration.